src/components/Forms/CreatePostForm/CreatePostForm.js

```diff
@@ -0,0 +1,141 @@
+import React, { useState } from 'react';
+import Button from '@material-ui/core/Button';
+import { Redirect } from 'react-router-dom';
+import cookie from 'react-cookies';
+import {
+  TextField,
+  Dialog,
+  DialogActions,
+  DialogContent,
+  DialogContentText,
+  DialogTitle,
+} from '@material-ui/core';
+import config from '../../../config';
+import { serviceRequest } from '../../../services/serviceRequest';
+import { ErrorMessage } from '../../ErrorMessage';
+
+const path = '/api/v1/user/createpost';
+const domain = config.apiDomain;
+
+function preparePayload(method, data) {
+  const token = cookie.load('jwtToken');
+  const requestHeader = { Authorization: token };
+  const url = `https://${domain}${path}`;
+  return {
+    method,
+    url,
+    headers: requestHeader,
+    data,
+  };
+}
+
+export const CreatePostForm = () => {
+  const [open, setOpen] = useState(false);
+  const [formData, setForm] = useState({ label: 'general', title: null, content: null });
+  const [isError, setIsError] = useState(false);
+  const [isValidPost, setIsValidPost] = useState(false);
+  const [Url, setUrl] = useState('');
+  const [errorMsg, setErrorMsg] = useState('Please make sure all fields are complete');
+
+  const handleChange = (event) => {
+    const updateForm = { ...formData };
+    updateForm[event.target.name] = event.target.value;
+    setForm(updateForm);
+  };
+
+  const handleClickOpen = () => {
+    setOpen(true);
+  };
+
+  const handleClose = () => {
+    setOpen(false);
+    setErrorMsg('');
+  };
+
+  const handleCreate = async (e) => {
+    e.preventDefault();
+    if (!formData.label || !formData.title || !formData.content) return;
```

Comment:
It would be useful if you display a message to the user to tell him.her that this field needs to be filled out to submit otherwise cancel

Reply:
The `required` attribute on each `TextField` would have a pop up message showing up if any field is not yet filled out.